### PR TITLE
Add coding guideline for optional fields.

### LINKED ISSUE #9478: [confighttp] Review usage of 'pointer to type'
There are a few fields in `confighttp` structs that use a pointer to int instead of an int. This is an unusual pattern in configuration from components, so I think we should review why we use it and whether it is justified.

To fix this issue we need to:

- Define when we can use 'pointer to type' in configuration vs just type and relying on the zero value
- Review the fields on `confighttp` based on this.

### PR DESCRIPTION
**Pull request by mackjmr**
#### Description
Specify to use pointers to represent optional fields. This is a follow up to: https://github.com/open-telemetry/opentelemetry-collector/issues/9478.

#### Link to tracking issue
Fixes #9478

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -294,6 +294,10 @@ To guarantee backward-compatible behavior, all configuration packages should sup
 
 Users should always initialize the config struct with this function and overwrite anything as needed.
 
+### Optional Fields
+
+Use pointers to represent optional fields in configuration to distinguish between unset values and zero values.
```
[the remark this answers is not in the document]
Comment (mackjmr) [reply CONTRIBUTING.md]: updated 👍

**Review by songy23 — APPROVED** [review]
LGTM

**Review by mx-psi — COMMENTED** [review]
After discussing with Evan, we may want to do a PoC of the hook that would be needed to support `Optional` to have a more concrete idea of the complexity we would be adding with this alternative.

**Comment by mackjmr** [comment]
@evan-bradley @yurishkuro @mx-psi  I've taken a stab at this in #11409, can you please take a look ? I may be missing some edge cases.

**Pull request closed without merging**